module/applications/item/item-sheet.mjs

```diff
@@ -109,4 +118,39 @@ export default class ItemSheetEd extends ItemSheet {
     const effect = this.item.effects.get( li.dataset.itemId );
     return effect.sheet?.render( true );
   }
+
+  /* ----------------------------------------------------------------------- */
+  /*                Auto calculation for equipments weight                   */
+  /* ----------------------------------------------------------------------- */
+
+  /**
+   * Handle autorecalculation of physical items for actors, based on the namegiver modifier for weight.
+   */
+  async _onWeightCalculation( ) {
```

Comment:
Also, we should think about moving that whole function to the item document class. Since updating item data is its responsibility. Not the sheet. The listener should only call the changing method, refresh the sheet.

Reply:
done

Reply:
Streamlined in commit 74f3682a5f5384e2ae4986733dec0a073f2fde18.